stack-build

```diff
@@ -19,42 +32,51 @@ get_build_id () {
 }
 
 usage () {
-    echo "usage: stack-build -n name -r recipe-path -s system -b build-root [-d]"
+    echo "usage: stack-build -n name -r recipe-path -s system -b build-root -u uarch -m mount [-d]"
     echo ""
     echo "where:"
     echo "  name:        the name of the stack"
     echo "  system:      the cluster name \(one of balfrin, hohgant, clariden\)"
     echo "  recipe-path: the location of the recipe"
     echo '  build-root:  the root path in which to perform the build, e.g. /dev/shm/$USER'
+    echo "  uarch:       the target micro-architecture"
+    echo "  mount:       the mount point of the image"
     echo ""
     [[ "" == "$1" ]] && exit 0
     err "$1"
 }
 
-system="-"
 name="-"
+system="-"
 recipe_path="-"
 build_root="-"
+uarch="-"
+mount="-"
 spack_develop=""
 
-while getopts n:s:r:b:d flag
+while getopts n:s:r:b:u:m:d: flag
```

Comment:
This change makes the `-d` option expect an argument. This was likely un-detected because the option set up in `configuration.py` ends up being illegal (see comment above).